mmv1/templates/terraform/pre_update/alloydb_cluster.go.erb

```diff
@@ -3,6 +3,11 @@ if d.HasChange("cluster_type") && d.Get("cluster_type") == "SECONDARY" {
     return fmt.Errorf("Can not convert a primary cluster to a secondary cluster.")
 }
 
+// Restrict setting secondary_config if cluster_type is PRIMARY
+if d.Get("cluster_type") == "PRIMARY" && d.Get("secondary_config") != nil {
```

Comment:
I think we get back an empty `[]interface{}`, not `nil`! Terraform only returns `nil` for unrecognized fields, generally.

Reply:
Fixed